src/main.cpp

```diff
@@ -1007,9 +1089,23 @@ int main(int argc, char *argv[]) {
     for (size_t i = 0; i < AMDP_names.size(); i++)
     {
       std::string varname = "/housekeeping/amd/" + AMDP_names.at(i);
-      h5_write_buffer<cl_float>(out_name, varname.c_str(), amd_power0[i].data(), amd_power0[i].size(),
+      //cout<<"Power: "<< AMDP_names.at(i) <<endl;
+      h5_write_buffer<cl_float>(out_name, varname.c_str(), amd_power[i].data(), amd_power[i].size(),
       "Power in watt");
     }
+
+  }
+  if (amd_temp_rate > 0)
+  {
+      h5_write_buffer<double>(out_name, "/housekeeping/amd/temperature_time", amd_power_time.data(), amd_power_time.size(),
```

Comment:
Shouldn't this be `amd_temp_time` instead of `amd_power_time`?

Reply:
It should, if two separate threads with different sampling rates are used, but unfortunately, this does not work. So right now it is more resource efficient to use one thread with one time vector. The problem is that if both temperature and power logging are enabled, the driver only returns power. Separately it works.

Reply:
Okay. Could you please add a corresponding remark in the source code and file an issue?